cmake/cxx_standard.cmake

```diff
@@ -12,3 +12,7 @@ else ()
     message("Test is using C++20")
     set(CMAKE_CXX_STANDARD 20)
 endif ()
+
+if (CMAKE_SYSTEM_NAME STREQUAL "Linux")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
```

Comment:
CMake has a FindThreads module that might be more robust. we could also scope the `Threads::Threads` dependency to a particular target.
https://stackoverflow.com/a/29871891
https://cmake.org/cmake/help/latest/module/FindThreads.html